lib/cocoapods/installer/target_installer/pod_target_installer.rb

```diff
@@ -66,9 +67,13 @@ def add_files_to_build_phases
           native_target.add_file_references(header_file_refs) do |build_file|
             # Set added headers as public if needed
             if target.requires_frameworks?
+              build_file.settings ||= {}
               if public_headers.include?(build_file.file_ref.real_path)
-                build_file.settings ||= {}
                 build_file.settings['ATTRIBUTES'] = ['Public']
+              elsif private_headers.include?(build_file.file_ref.real_path)
+                build_file.settings['ATTRIBUTES'] = ['Private']
+              else
+                build_file.settings['ATTRIBUTES'] = ['Project']
```

Comment:
@neonichu this only covers the case in which a user specifies both public & private headers, and there are headers which are included in neither -- those become the project headers.